### PR TITLE
Add input_type option

### PR DESCRIPTION
**Pull request by billytrend-cohere**
Add input_type option to support [embed v3 models](https://txt.cohere.com/introducing-embed-v3/)

### CODE REVIEW

File: libs/langchain/langchain/embeddings/cohere.py [hunk 1]
```diff
@@ -87,7 +98,7 @@ def embed_documents(self, texts: List[str]) -> List[List[float]]:
             List of embeddings, one for each text.
         """
         embeddings = self.client.embed(
-            model=self.model, texts=texts, truncate=self.truncate
+            model=self.model, texts=texts, truncate=self.truncate, input_type=self.input_type
```
Comment (baskaryan) [inline libs/langchain/langchain/embeddings/cohere.py:101]: will it break things for v2 models to pass input_type?

Comment (billytrend-cohere) [reply libs/langchain/langchain/embeddings/cohere.py]: it will just be ignored, have tested

**Comment by kacperlukawski** [comment]
@billytrend-cohere Shouldn't we use a different value of `input_type` for documents and queries?

**Comment by billytrend-cohere** [comment]
You're right, updated to support user override but also sensible defaults. Thanks!

**Comment by billytrend-cohere** [comment]
Actually sorry I don't think we should merge this. Thinking about it, I don't think we should try provide a default value for this, else we end up with the same issue as the model default as we upgrade. Will update PR now

**Pull request closed without merging**